File: ch2/yojkim/ch2/dollar_test.go
package ch2

import (
	"github.com/stretchr/testify/assert"
	"testing"
)

func TestMultiplication(t *testing.T) {
	five := NewDollar(5)
	product := five.Times(2)
	assert.Equal(t, 10, product.Amount)
	product = five.Times(3)
	assert.Equal(t, 15, product.Amount)
}

File: ch2/gunddol/src/ch2/dollar_Test.java
package ch2;

import org.junit.Test;

import static org.junit.Assert.*;

public class dollar_Test {
    @Test
    public void testMultiplication() {
        dollar five = new dollar(5);
        dollar product = five.times(2);
        assertEquals(10, product.amount);
        product = five.times(3);
        assertEquals(15, product.amount);
    }
}